src/infra/auth/jwt.ts

```diff
@@ -64,26 +66,29 @@ let $getUserTokenDataCallback: GetUserTokenDataFn = async (
 			options: {
 				$select: tokenFields,
 			},
-		}) as Promise<AnyObject>,
+		}) as Promise<PickDeferred<DbUser, (typeof tokenFields)[number]>>,
 		permissions.getUserPermissions(userId, tx),
 	]);
 	if (!userData || !permissionData) {
 		throw new Error('No data found?!');
 	}
-	const newTokenData: Partial<User> = _.pick(userData, tokenFields);
+	const newTokenData = {
+		..._.pick(userData, tokenFields),
+		actor: userData.actor.__id,
+	};
 
-	const tokenData = {
+	const tokenData: TokenUserPayload = {
 		...existingToken,
 		...newTokenData,
 		permissions: permissionData,
-	} as User;
+	};
 
 	if (!Number.isFinite(tokenData.authTime!)) {
 		tokenData.authTime = Date.now();
 	}
 
 	// skip nullish attributes
-	return _.omitBy(tokenData, _.isNil) as User;
+	return _.omitBy(tokenData, _.isNil) as TokenUserPayload;
```

Comment:
We do need it atm, since `_.omitBy` returns `Dictionary<any>`.
The ideal typings would need to convert the nulls to optional props like v
https://stackoverflow.com/a/72165304
but since we atm only use the result to pass it into `jsonwebtoken.sign` I didn't invest more time on this.